### ISSUE TITLE
Support Non-json stdin unstructured log messages

### ISSUE THREAD

**Issue by StevenACoffman** [fluent/fluent-bit#220]
Currently, the stdin plugin only supports json format. It works very well for json, so thank you!

However, the recently released [parser feature](http://fluentbit.io/documentation/0.11/getting_started/parser.html) was intended to support converting unstructured log messages into structured ones. As it stands, it is difficult to fully utilize this feature.

Currently, if you have try to input an unstructured, non-json log message such as one piped from Apache to the stdin plugin you will only get:
> `STDIN data incomplete, waiting for more data...`
The input step occurs before you have the opportunity to run a parser, so it is too late to convert into a structured format.

For Apache, there are other workarounds: 
1. Use tail input with logrotate
2. Customize the LogFormat to emit json (see [here](https://gist.github.com/vmadman/5472166) and [here](http://blog.client9.com/2013/08/19/json-log-format-for-apache.html))
3. Pipe to a series of hacks (e.g. sed) to encode the message as json

These only work for Apache, and there are other reasons these workarounds are not preferred.

It would be super nice if there was an input plugin for non-json, unstructured stdin so we could pipe to a parser plugin. Regardless, thanks for doing awesome stuff!

**Comment by edsiper** [comment]
@StevenACoffman 

I pushed a test/dev branch called stdin_parse, would you please checkout that one and give it a try ?, now is possible to define a parser for stdin plugin. I am sure it needs some tweaks but it looks functional.

please send me some feedback.

**Comment by StevenACoffman** [comment]
@sellers maybe we can check this out monday.
@edsiper Will do! Thanks!

**Comment by StevenACoffman** [comment]
@edsiper This looks really good!
I looked at [parsers.conf](https://github.com/fluent/fluent-bit/blob/master/conf/parsers.conf) and wanted to use the apache2 and apache_error. I pulled and built from your branch and it seems like it works great!

As a test, I copied my apache_access.log and ran this:
`cat apache_access.log | ./bin/fluent-bit -v -i stdin -p Parser=apache2 --parser=../conf/parsers.conf -o stdout`

Similarly, after having installed this, having apache configured to use:
`CustomLog "|/bin/fluent-bit -v -i stdin -p Parser=apache2 --parser=/opt/build/fluent-bit/conf/parsers.conf -o stdout"`
Seems to work well!

**Comment by edsiper** [comment]
@StevenACoffman 

awesome :) , thanks for testing.

Does your apache_access.log have some break line (\n) ?, not sure if would be good to have support for that.

**Comment by edsiper** [comment]
@StevenACoffman 

I've merged the changes into master branch, so they will be available on 0.12 major release. I am closing this issue for now but feel free to comment out here if you found any problem.

thanks again.

**Comment by StevenACoffman** [comment]
Yeah, mine does have Unix line endings (`\n` or LF) line endings and it worked great as is. Here's an example block:
```
127.0.0.1 - - [03/Apr/2017:15:36:41 +0000] "GET /server-status?auto HTTP/1.1" 200 499 "-" "collectd/5.4.0.git"
127.0.0.1 - - [03/Apr/2017:15:36:51 +0000] "GET /server-status?auto HTTP/1.1" 200 498 "-" "collectd/5.4.0.git"
127.0.0.1 - - [03/Apr/2017:15:37:01 +0000] "GET /server-status?auto HTTP/1.1" 200 498 "-" "collectd/5.4.0.git"
127.0.0.1 - - [03/Apr/2017:15:37:11 +0000] "GET /server-status?auto HTTP/1.1" 200 499 "-" "collectd/5.4.0.git"
127.0.0.1 - - [03/Apr/2017:15:37:21 +0000] "GET /server-status?auto HTTP/1.1" 200 497 "-" "collectd/5.4.0.git"
127.0.0.1 - - [03/Apr/2017:15:37:31 +0000] "GET /server-status?auto HTTP/1.1" 200 499 "-" "collectd/5.4.0.git"
127.0.0.1 - - [03/Apr/2017:15:37:41 +0000] "GET /server-status?auto HTTP/1.1" 200 495 "-" "collectd/5.4.0.git"
```
For possible others reference, you can see the line endings in a file using `cat -e`. This displays Unix line endings (`\n` or LF) as `$` and Windows line endings (`\r\n` or CRLF) as `^M$`.
